index.html

```diff
@@ -232,39 +235,393 @@ <h3>Create</h3>
+the resulting output.
+          </p>
+
+          <ol class="algorithm">
+            <li>
+Initialize <em>verification method</em> to an empty object.
+            </li>
+            <li>
+Set <em>multicodec value</em> and <em>raw public key bytes</em> to the result
+of passing <em>multibase value</em> and <em>options</em> to
+<a href="#decode-public-key-algorithm"></a>.
+            </li>
+            <li>
+Ensure the proper key length of <em>raw public key bytes</em> based on the
+<em>multicodec value</em> table provided below:
+              <table class="simple">
+                <thead>
+                  <th>Multicodec hexadecimal value</th>
+                  <th>public key byte length</th>
+                  <th>Description</th>
+                </thead>
+
+                <tbody>
+                  <tr>
+                    <td>`0xe7`</td>
+                    <td>33 bytes</td>
+                    <td>secp256k1-pub - Secp256k1 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0xec`</td>
+                    <td>32 bytes</td>
+                    <td>x25519-pub - Curve25519 public key</td>
+                  </tr>
+                  <tr>
+                    <td>`0xed`</td>
+                    <td>32 bytes</td>
+                    <td>ed25519-pub - Ed25519 public key</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1200`</td>
+                    <td>33 bytes</td>
+                    <td>p256-pub - P-256 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1201`</td>
+                    <td>?? bytes</td>
+                    <td>p384-pub - P-384 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1202`</td>
+                    <td>?? bytes</td>
+                    <td>p521-pub - P-521 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1205`</td>
+                    <td>?? bytes</td>
+                    <td>
+rsa-pub - RSA public key. DER-encoded ASN.1 type RSAPublicKey according to
+IETF RFC 8017 (PKCS #1)
+                    </td>
+                  </tr>
+                </tbody>
+              </table>
+If the byte length of <em>raw public key bytes</em> does not match the expected
+public key length for the associated <em>multicodec value</em>, an
+INVALID_PUBLIC_KEY_LENGTH error MUST be raised.
+            </li>
+            <li>
+Set the <em>verification method</em> `id` value by concatenating
+<em>identifier</em>, a hash character (`#`), and the <em>multicodec value</em>.
+            </li>
+            <li>
+Set the <em>key encoding type</em> value to the <em>options</em> `type` value
+if it is set, or `Multikey` if it is not set.
+            </li>
+            <li>
+If <em>options</em> `allowExperimentalPublicKeyTypes` is not set to `true` and
```

Comment:
this reads a bit weird, but you need to `allowExperimentalPublicKeyTypes` in order to present the key with something other than `MultiKey`, `JsonWebKey` or `Ed25519VerificationKey2020`... so for example, to support backwards compatibility with `Ed25519VVerifcaitonKey2018`... I think this option approach makes sense.

Reply:
Yes, that's the intent of the language... and yes, it reads weird -- I'll try to make another pass at the language to make it easier to read. I was playing around w/ the use of abstract variables, but then had to use concrete ones (like `allowExperimentalPublicKeyTypes`)... so right now, it's a mish-mash of both, which I don't think is helping understandability. I'll try to go back and look at what some of the HTML5 and JSON-LD 1.1 Algorithms did here.

Reply:
I got it, just had to read it a few times, you can resolve this comment.

Reply:
Added some language in 2bcf5d7d2a50f92396050b7a514dd97f8d67d9a6 in an attempt to make it easier to read. Still don't think it helped much. 

I also tried moving to easier to read variable names using dot-notation like the Fetch spec does and I think that helped a bit. Those changes are in 63f46992c51bbb2de022e720f6f82632bcb831df.